crates/moon/src/frontend.rs

```diff
@@ -12,6 +13,7 @@ pub struct Frontend {
 impl Default for Frontend {
     fn default() -> Self {
         Self {
+            lang: Cow::from("en"),
```

Comment:
What do you think about `lang: Lang`, where the default one would be `Lang::En` or `Lang::EnUs` (I don't know if just `en` is a valid lang value.)

`Lang` would be similar to `Tag`: https://github.com/MoonZoon/MoonZoon/blob/457b8d17f8ec6ee33565a90c1e7e013929d0e31c/crates/zoon/src/element/ability/choosable_tag.rs.
And we could generate `Lang` variants with a macro or/and we can load variants from a file.

The idea is to mitigate app developer typos and make sure the developer don't have to think whether `en` or `EN` or `en-Us` or `EN-US` is a correct value.

Reply:
I tried to make choosable_lang just like the choosable_tag module, but to be honest I am having trouble figuring out where this module file would be placed so it can be used in both "frontend.rs" in the moon crate and in the has_lang trait in the zoon crate. I can push the commit if you'd like but as of now choosable_lang is only in zoon. Also the choosable_lang module is super long because I ended up just copying all langs and their attribute values off wikipedia. Should've imported from a file like you suggested

Reply:
> I tried to make choosable_lang

You can leave the old name (`HasLang`) and structure I think. I've sent the link to `ChoosableTag` just for inspiration.

> where this module file would be placed

Very good question. 

There is the shared `moonlight` crate between frontend and backend but its primary purpose is to help with communication between FE and BE so I probably wouldn't add `lang` into it because it doesn't help with communication at all now. However, I can imagine we will want to allow to choose the server `lang` value from the frontend side - see for example https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language - then it would become a part of the communication basically.

Another option would be to create a new crate in the MZ repo - see for example https://docs.rs/rocket_lang/0.0.2/rocket_lang/enum.LangCode.html (but there is also some Rocket-specific "mess" so we can't use it directly unfortunately).

I would probably create a new crate. It would allow us to experiment with it and maybe publish later as a standalone crate so other devs can use it even without a MZ-related project.

Note: I'm not sure what ISO/standard would be the best one. That Rocket crate uses [ISO 639-1](https://en.wikipedia.org/wiki/ISO_639-1). Or we can use its newer version `-2` or [-3](https://en.wikipedia.org/wiki/ISO_639-3) or other standard or multiple ones to cover `aria`, `accept-language` and other eventual needs.

Also don't be afraid to start with a hardcoded enum with only some variants just to make sure everything works and then we can add the remaining variants and generators from files or even auto-downloaders from specs and similar fancy things. Just step by step :)

Reply:
I made `lang ` into a crate so now it can be tested with both backend and frontend. Tested the changes in the paragraph example project. I only started learning Rust in February so sorry if I'm making noob errors :'D

Reply:
>  I only started learning Rust in February so sorry if I'm making noob errors :'D

The code is clean / simple / without typos and there are either pretty obvious Rust-specific problems that you do only once and you'll never do it again or project-specific cosmetic problems that aren't real problems actually. Honestly, it's better than many PRs I had the "honor" to review from senior devs 😅